rasa_core/domain.py

```diff
@@ -416,7 +416,27 @@ def load(cls, filename, action_factory=None):
         )
 
     @classmethod
-    def validate_domain_yaml(cls, filename):
+    def load_from_yaml(cls, yaml, action_factory=None):
```

Comment:
can `load` use `load_from_yaml` once it read the yaml from the file as well? want to avoid the duplication